File: 18-encryption/cpp/main.cpp
// g++ -std=c++11 -Wall *.cpp -o output
#include "rot13.h"
#include <iostream>
#include <fstream>

int main(int argc, char* argv[]) {
  if(argc == 0) {
    std::cout << "Provide a file to convert." << std::endl;
    return 0;
  }

  for (int i = 0; i < argc; i++) {
    std::cout << argv[i] << std::endl;
  }

  std::string filename = argv[0];
  if(std::fstream is(filename)) {
    auto size = is.tellg(); // wtf
    std::string str(size, '\0');
    is.seekg(0);
    is.read(&str[0], size);
    std::cout << str << "\n";
  } else {
    std::cerr << "Couldn't open " << filename << std::endl;
  }

  std::cout << rot13("Hello World!") << std::endl;

  return 0;
}

File: 18-encryption/py/main.py
"""Converts a file using a rot13 algorithm"""

import sys
import rot13

def main():
    """Run the programme"""

    if len(sys.argv) < 2:
        print("Provide the name of a file to convert")
        sys.exit()

    filename = sys.argv[1]

    try:
        file = open(filename, "r+")
        text = file.read()
        file.seek(0) # go to start of file
        file.write(rot13.convert(text))
        file.close()
        print("Converted {}".format(filename))
    except IOError:
        print("Can't open file")

if __name__ == '__main__':
    main()
